aten/src/ATen/native/Loss.cpp

```diff
@@ -351,38 +391,6 @@ Tensor soft_margin_loss(
   return output;
 }
 
-Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
-  TORCH_CHECK(beta >= 0, "smooth_l1_loss does not support negative values for beta.")
-  if (beta == 0) {
-      return at::native::l1_loss(input, target, reduction);
-  }
-  Tensor loss;
-  auto iter = TensorIterator::borrowing_binary_op(loss, input, target);
-  smooth_l1_stub(iter.device_type(), iter, beta);
-  return apply_loss_reduction(iter.output(), reduction);
-}
-
-Tensor& smooth_l1_loss_out(const Tensor& input, const Tensor& target, int64_t reduction, double beta, Tensor& result) {
-  TORCH_CHECK(beta >= 0, "smooth_l1_loss does not support negative values for beta.")
-  if (beta == 0) {
-      return at::native::l1_loss_out(input, target, reduction, result);
-  }
-  if (reduction != Reduction::None) {
-    Tensor loss;
-    auto iter = TensorIterator::borrowing_binary_op(loss, input, target);
-    smooth_l1_stub(iter.device_type(), iter, beta);
-    if (reduction == Reduction::Mean) {
-      at::mean_out(result, iter.output(), 0);
```

Comment:
yeah I think you're right, this was just broken. Tested:
```
>>> a = torch.ones(2, 2)
>>> b1 = torch.ones(())
>>> b1.size()
torch.Size([])
>>> torch._C._nn.smooth_l1_loss(a, a, beta=1.5, reduction=1, out=b1)
tensor([0., 0.])
```

It probably isn't hit anywhere by most people (or in tests) because most people use the loss function through the `torch.nn.functional()` API, which doesn't call the `out=` variant.